Sources/PandaSDK/Extension+StoreKit.swift

```diff
@@ -0,0 +1,170 @@
+//
+//  Extension+StoreKit.swift
+//  PandaSDK
+//
+//  Created by Andrew Skrypnyk on 23.07.2020.
+//
+
+import Foundation
+import StoreKit
+
+extension SKProduct {
+    
+    func localizedString(with offerID : String? = nil) -> String {
+        
+        guard let offer = offerID else {
+            return localizedPrice()
+        }
+        
+        guard #available(iOS 12.2, *) else {
+            print("Minimum iOS 12.2 required for offerID [\(offer)] at product [\(productIdentifier)]")
+            return ""
+        }
+        
+        guard let discount = discounts.first(where: {$0.identifier == offerID!}) else {
+            print("Couldn't find [\(offer)] at [\(productIdentifier)]")
+            return ""
+        }
+        
+        return localizedDiscountPrice(discount: discount)
+    }
+    
+    private func unitStringFrom(periodUnit : SKProduct.PeriodUnit) -> String {
+        var unit = ""
+        switch periodUnit {
+        case .day:
+            unit = "day"
+        case .week:
+            unit = "week"
+        case .month:
+            unit = "month"
+        case .year:
+            unit = "year"
+        default:
+            break
+        }
+        return unit
+    }
+    
+    @available(iOS 12.2, *)
+    private func promoParameters(discount : SKProductDiscount) -> [String : Any] {
+        
+        let periods_count = discount.numberOfPeriods
+        
+        let unit_count = discount.subscriptionPeriod.numberOfUnits
+        
+        var mode :String = ""
+        switch discount.paymentMode {
+        case .payUpFront:
+            mode = "pay_up_front"
+        case .payAsYouGo:
+            mode = "pay_as_you_go"
+        case .freeTrial:
+            mode = "trial"
+        default:
+            break
+        }
+        
+        let unit = unitStringFrom(periodUnit: discount.subscriptionPeriod.unit)
+        
+        return ["unit" : unit, "units_count" : unit_count, "periods_count" : periods_count, "mode" : mode, "price" : discount.price.floatValue, "offer_id" : discount.identifier ?? ""]
+    }
+    
+    @available(iOS 12.2, *)
```

Comment:
but in specification min version is 12, maybe we can update version in spec?

Reply:
we still should use 11.2 as min spec version ;(